src/MIDebugEngine/Engine.Impl/Variables.cs

```diff
@@ -428,16 +438,35 @@ private string StripFormatSpecifier(string exp, out string formatSpecifier)
             }
 
             // array with static size
-            // TODO: could return '(T(*)[n])(exp)' but requires T
             var m = Regex.Match(trimmed, @"^\[?(\d+)\]?$");
             if (m.Success)
             {
+                string count = m.Groups[1].Value; // (\d+) capture group
+                string expr = exp.Substring(0, lastComma);
+
                 if (_engine.DebuggedProcess.MICommandFactory.Mode == MIMode.Gdb)
-                    return "*" + exp.Substring(0, lastComma) + "@" + trimmed;  // this does not work for lldb
+                {
+                    // return *<expression>@<count> which is only supported in GDB
+                    return FormattableString.Invariant($"*{expr}@{count}");
```

Comment:
Does this work with more complex expressions?
Parens around expr might be needed.

Reply:
Also I think this syntax works for dynamic sizes too, see L472.